.github/workflows/doc_pages.yaml

```diff
@@ -23,7 +23,10 @@ jobs:
           version: 1.3.2
 
       - name: install
-        run: poetry install
+        run: poetry install -E black
```

Comment:
Is `black` needed for the docs?

Reply:
it's highly desirable for the docs! 

i think it's in general a good move to not try and nail the formatting in the templates and generators, and just to get them into syntactically correct python and let black handle the rest.

so if `black` is present, we use it - https://github.com/linkml/linkml/blob/dd1ce611f83b00484929abdec157c28665d2fe7e/linkml/generators/pydanticgen/template.py#L85

but it also works without it. 

tested here: https://github.com/linkml/linkml/blob/pydanticgen_array_reps/tests/test_generators/test_pydanticgen.py#L1602-1691

(test was actually a pain in the ass to write, it's actually pretty hard to mock a module not being present once everything has already imported it, but i [made it work](https://github.com/linkml/linkml/blob/dd1ce611f83b00484929abdec157c28665d2fe7e/tests/conftest.py#L211-L252) lol)